lang/py/src/avro/io.py

```diff
@@ -583,16 +699,36 @@ def read_data(self, writers_schema, readers_schema, decoder):
     elif writers_schema.type == 'string':
       return decoder.read_utf8()
     elif writers_schema.type == 'int':
-      return decoder.read_int()
+      if hasattr(writers_schema, 'logical_type'):
+        if writers_schema.logical_type == constants.DATE:
+          return decoder.read_date_from_int()
+        elif writers_schema.logical_type == constants.TIME_MILLIS:
+          return decoder.read_time_millis_from_int()
+        else:
+          fail_msg = "Cannot read unknown schema type: %s logicalType %s" % (writers_schema.type, writers_schema.logical_type)
```

Comment:
This may not be what it should do. The spec specifies: `Language implementations must ignore unknown logical types when reading, and should use the underlying Avro type. If a logical type is invalid, for example a decimal with scale greater than its precision, then implementations should ignore the logical type and use the underlying Avro type.`

Could you also check with Java implementation to see how they handle such case currently?